src/services/upload-form-model.ts

```diff
@@ -44,6 +52,10 @@ export class UploadFormModel {
     const submission = this.submissions.find((s) => s.id === id);
     if (!submission || !submission.uploadPolicy) return;
 
-    return await this.gateway.upload(file, submission.uploadPolicy);
+    await this.s3.upload(file, submission.uploadPolicy);
+
+    await this.gateway.updateDocumentSubmission(this.evidenceRequestId, id, {
```

Comment:
Given that we're now also updating the document submission status we could rename `uploadFile` to `uploadFileAndUpdateDocumentSubmission`, or move the call to `updateDocumentSubmission` into `handleSubmit`